spec/validators/root_protection_validator_spec.rb

```diff
@@ -3,38 +3,38 @@
 class RootProtectionValidatable
   include ActiveModel::Validations
   validates_with RootProtectionValidator
-  attr_reader :parent
+  attr_reader :type
 
-  def initialize(parent)
-    @parent = parent
+  def initialize(type)
+    @type = type
   end
 end
 
 describe RootProtectionValidator do
-  subject { RootProtectionValidatable.new parent }
+  subject { RootProtectionValidatable.new type }
 
   context 'A root node exists' do
     let!(:root_node) { Fabricate(:root_node) }
 
-    context 'with a parent' do
-      let(:parent) { 'something' }
-      it { is_expected.to be_valid }
+    context 'with a root node' do
+      let(:type) { 'RootNode' }
+      it { is_expected.not_to be_valid }
     end
 
-    context 'without a parent' do
-      let(:parent) { nil }
-      it { is_expected.not_to be_valid }
+    context 'without a root node' do
+      let(:type) { 'something' }
+      it { is_expected.to be_valid }
     end
   end
 
   context 'No root node exists' do
-    context 'with a parent' do
-      let(:parent) { 'something' }
+    context 'with a root node' do
+      let(:type) { 'RootNode' }
       it { is_expected.to be_valid }
     end
 
     context 'without a parent' do
```

Comment:
Yes the language is a bit confusing, as there are two contexts - the first context talking about what's already existing, and the second talking about what's being created. I might rename to make clearer.